### PR TITLE
Upd/middleware

### PR DESCRIPTION
**Pull request by fanglinfang**

### CODE REVIEW

File: uw_oidc/middleware.py [hunk 1]
```diff
@@ -25,34 +29,35 @@ def process_view(self, request, view_func, view_args, view_kwargs):
...
             if (request.user.is_authenticated and
                     self.TOKEN_SESSION_KEY in request.session):
-                # The user is authenticated with a token in session, but the
-                # request does not contain a token, the session is invalid.
-                del request.session[self.TOKEN_SESSION_KEY]
+                # The session was established based on a valid token
+                # but the request has no token, revoke the existing session.
+                logger.error("Revoke token based session: {}".format(
+                    request.user.get_username()))
                 auth.logout(request)
```
Comment (fanglinfang) [inline uw_oidc/middleware.py]: logout will flush all session data.

File: README.md [hunk 2]
```diff
@@ -20,7 +20,6 @@ TOKEN_ISSUER = ''
 TOKEN_AUDIENCE = ''
 
 # Specifies the allowed validity window to accommodate clock skew
-# between the given issue time and expiration time (in minutes)
-# that the ID token is within
-TOKEN_LEEWAY = ''
+# between the given expiration time of the ID token
+TOKEN_LEEWAY = seconds
```
Comment (jlaney) [inline README.md:24]: This is called UW_TOKEN_LEEWAY in id_token.py.

File: uw_oidc/middleware.py [hunk 3]
```diff
@@ -25,34 +29,35 @@ def process_view(self, request, view_func, view_args, view_kwargs):
...
-                        auth.login(request, user)
-                        request.session[self.TOKEN_SESSION_KEY] = token
-
+                user = auth.authenticate(request, remote_user=username)
+                if user:
+                    # User is valid.  Set request.user and persist user
+                    # in the session by logging the user in.
+                    auth.login(request, user)
+                    request.session[self.TOKEN_SESSION_KEY] = token
+                    logger.info(json.dumps(
```
Comment (jlaney) [inline uw_oidc/middleware.py:48]: I don't see the value of this logging.

File: uw_oidc/middleware.py [hunk 4]
```diff
@@ -25,34 +29,35 @@ def process_view(self, request, view_func, view_args, view_kwargs):
...
                                     reason='Invalid token: {}'.format(ex))
         else:
             if (request.user.is_authenticated and
                     self.TOKEN_SESSION_KEY in request.session):
-                # The user is authenticated with a token in session, but the
-                # request does not contain a token, the session is invalid.
-                del request.session[self.TOKEN_SESSION_KEY]
+                # The session was established based on a valid token
+                # but the request has no token, revoke the existing session.
+                logger.error("Revoke token based session: {}".format(
```
Comment (jlaney) [inline uw_oidc/middleware.py:59]: Logging an error here will send email to the django admins if they do not add a logging handler in settings.  My preference would be to remove logging from this library.

Comment (fanglinfang) [reply uw_oidc/middleware.py:59]: These logging will be useful in the first stage of the integration testing. I have a checklist item to check the logging and reduce them afterwards.

File: README.md (hunk 2 above)
Comment (fanglinfang) [reply README.md]: Ah, thanks!

File: uw_oidc/middleware.py (hunk 3 above)
Comment (fanglinfang) [reply uw_oidc/middleware.py:48]: It is helpful from MyUW's stand point to know when a token based session is initiated.

**Pull request merged**